File: frontend/src/model/LocationReturn.ts
export type LocationReturn = {
    address:{
        city:string | undefined;
        country_code:string | undefined;
        postcode:string | undefined;
        road:string| undefined;
        state:string| undefined;
        suburb:string| undefined;
    } |undefined;
    osm_id:number | undefined;
    display_name:string | undefined;
    lat:string | undefined;
    lon:string | undefined;
}

File: backend/src/main/java/de/neuefische/backend/model/LocationReturn.java
package de.neuefische.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document("locations")
public class LocationReturn {
    @Id
    private String osm_id;
    private double lat;
    private double lon;
    private String display_name;
    private AddressDTO address;
}
